docs/source/api/algorithms/ensembles/decision-forest.rst

```diff
@@ -102,6 +102,25 @@ splitter_mode::random
    and selects the best feature in terms of impurity
    computed for that random split from the feature subsets.
 
+`engine_type`
+~~~~~~~~~~~~~
+
+engine_type::mt2203
+   The engine method is mt2203 - Mersenne Twister engine with specific optimizations for 
+   parallel environments.
+
+engine_type::mcg59
+   The engine method is mcg59 - Multiplicative congruential generator with a modulus of \(2^{59}\).
+
+engine_type::mt19937
+   The engine method is mt19937 - Standard Mersenne Twister engine with a period of \(2^{19937} - 1\).
+
+engine_type::mrg32k3a
+   The engine method is mrg32k3a - Combined multiple recursive generator with a period of \(2^{191}\).
+
+engine_type::philox4x32x10
+   The engine method is philox4x32x10 - Counter-based RNG engine optimized for parallel computations.
```

Comment:
It would be better to move the descriptions to .hpp file and here just add:

```suggestion
`engine_type`
~~~~~~~~~~~~~

.. onedal_enumclass:: oneapi::dal::engine_type
```